domain/cloudimagemetadata/state/state_test.go

```diff
@@ -0,0 +1,542 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state_test
```

Comment:
```suggestion
package state
```

We're not winning anything by being external testing.

Reply:
I tend to prefer external testing, to be able to spot interface leaks more easily. But ok.